packages/web/src/pages/userprofile.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { NextPage } from 'next';
+import { MarketingLayout } from '../components/Layout';
+import queryString from 'query-string';
+import { UserProfileLayout, UserProfileData } from '../components/Layout/UserProfileLayout';
+import { Alert, AlertIcon } from '@chakra-ui/alert';
+
+const UserProfile: NextPage = () => {
+    const [user, setUser] = React.useState<UserProfileData>({name: "", hireable: false, purpose: ""});
+    const [isValidUser, setIsValidUser] = React.useState<boolean>(false); // Flag used to catch whether id entered is in database
+
+    React.useEffect(() => {
+        const fetchStatus = async () => {
+
+            // Assuming that you need a query parameter to see a specific user's profile page (/userprofile?id=#)
+            // Concern?: Goes this route when there are additional parameters after id (/userprofile?id=#&key=value)
+            if(!isNaN(Number(queryString.parse(location.search).id)))
+            {
+                const userAPIQuery: string = "/api/users/" + queryString.parse(location.search).id;
+                
+                // If user id doesn't exist in database, throw the 400 error from users API
+                // Concern?: Catch block has nothing since isValidUser is by default set to false
```

Comment:
In this case, you can use the catch block to log an error if the userAPIQuery is invalid and causes the fetch to return null/undefined, this would cause the `res.json` to throw a null pointer expectation because json won't be defined.